x-pack/plugins/maps/public/shared/layers/vector_layer.js

```diff
@@ -349,45 +349,32 @@ export class VectorLayer extends AbstractLayer {
     }
   }
 
-  _joinToFeatureCollection(sourceResult, joinState, updateSourceData) {
-    if (!sourceResult.refreshed && !joinState.dataHasChanged) {
-      //no data changes in both the source data or the join data
-      return false;
-    }
-    if (!sourceResult.featureCollection || !joinState.propertiesMap) {
-      //no data available in source or join (ie. request is pending or data errored)
-      return false;
-    }
-
-    //all other cases, perform the join
-    //- source data changed but join data has not
-    //- join data changed but source data has not
-    //- both source and join data changed
-    const updatedFeatureCollection = joinState.join.joinPropertiesToFeatureCollection(
-      sourceResult.featureCollection,
-      joinState.propertiesMap);
-
-    updateSourceData(updatedFeatureCollection);
-
-    return true;
-  }
-
-  async _performJoins(sourceResult, joinStates, updateSourceData) {
-    const hasJoined = joinStates.map(joinState => {
-      return this._joinToFeatureCollection(sourceResult, joinState, updateSourceData);
-    });
-    return hasJoined.some(shouldRefresh => shouldRefresh === true);
-  }
-
-  async syncData({ startLoading, stopLoading, onLoadError, onRefreshStyle, dataFilters, updateSourceData }) {
+  async syncData({ startLoading, stopLoading, onLoadError, dataFilters, updateSourceData }) {
     if (!this.isVisible() || !this.showAtZoomLevel(dataFilters.zoom)) {
       return;
     }
+
     const sourceResult = await this._syncSource({ startLoading, stopLoading, onLoadError, dataFilters });
-    const joinResults = await this._syncJoins({ startLoading, stopLoading, onLoadError, dataFilters });
-    const shouldRefresh = await this._performJoins(sourceResult, joinResults, updateSourceData);
-    if (shouldRefresh) {
-      onRefreshStyle();
+    let sourceFeatureCollection = sourceResult.featureCollection;
```

Comment:
I think we might be able to simplify this. This is more an issue with the original implementation, than this PR. But it affects it.

`Join#joinPropertiesToFeatureCollection` shouldn't really need to return anything. It's really all side-effects. 

If so, I think we can get rid of the `let` assignment here, and not reassign to it later down. It might make the flow a little easier to understand. `updateSourceData` triggers the update due to store changes. `Join#joinPropertiesToFeatureCollection` doesn't really need to be concerned with this.